client/src/index.ts

```diff
@@ -470,3 +481,22 @@ export class XataError extends Error {
 }
 
 export type Links = Record<string, Array<string[]>>;
+
+const isBranchStrategyBuilder = (strategy: BranchStrategy): strategy is BranchStrategyBuilder => {
+  return typeof strategy === 'function';
+};
+
+const getBranchFromStrategy = async (param: BranchStrategyOption): Promise<BranchStrategyValue> => {
+  const strategies = Array.isArray(param) ? [...param] : [param];
+
+  const getBranch = async (strategy: BranchStrategy) => {
+    return isBranchStrategyBuilder(strategy) ? await strategy() : strategy;
+  };
+
+  for await (const strategy of strategies) {
+    const branch = await getBranch(strategy);
+    if (branch) return branch;
+  }
```

Comment:
I made the early return to avoid heavy tasks to be run if a valid strategy was already found. Not 100% sure yet if you are for/against this kind of practices.

Reply:
I like early returns 😄